2023-risks/m7- security-misconfiguration.md

```diff
@@ -0,0 +1,109 @@
+
+- Default settings not reviewed: Using default configurations without reviewing security settings, permissions and default credentials.
+- Lack of secure communication: Using unencrypted or weakly encrypted communication channels.
+- Weak or absent access controls: Allowing unauthorized access to sensitive functionality or data.
+- Failure to update or patch: Not applying necessary security updates or patches to the app or underlying components.
+- Improper storage of sensitive data: Storing sensitive data in plain text or weakly protected formats.
+
+To determine if your app is vulnerable to security misconfigurations, you should conduct a thorough security assessment, including code review, security testing, and configuration analysis.
+
+# How Do I Prevent Security Misconfigurations?
+
+Preventing security misconfigurations in mobile apps requires following secure coding and configuration practices. Here are some key prevention measures:
+
+- Secure default configurations: Ensure that default settings and configurations are properly secured and do not expose sensitive information or provide unnecessary permissions.
+- Default credentials: Refrain from using hardcoded default credentials.
+- Insecure permissions: Avoid storing application files with overly permissive permissions like world-readable and/or world-writable. 
+- Least privilege principle: Request only the permissions necessary for the proper functioning of the application
+- Secure network configuration: Disallow cleartext traffic and use certificate pinning when possible.
+- Disable Debugging: Disable debugging features in the production version of the app.
+- Disable backup mode (Android): By disabling backup mode on Android devices, you prevent the inclusion of app data in the device's backup, ensuring that sensitive data from the app is not stored in the device backup.
+
+
+# Example Attack Scenarios
+
+The following scenarios showcase security misconfigurations in mobile apps:
+
+**Scenario #1:** Insecure default settings.
+
+A mobile app is released with default settings that have weak security configurations enabled. This includes using insecure communication protocols, leaving default usernames and passwords unchanged, and not disabling debugging features in release builds. Attackers exploit these misconfigurations to gain unauthorized access to sensitive data or perform malicious actions.
+
+**Scenario #2:** Improper access controls.
+
+A mobile app lacks proper access controls, allowing unauthorized users to access sensitive functionality or data. This could lead to unauthorized access to user accounts, exposure of confidential information, or misuse of privileged actions.
+
+**Scenario #3:** Overly permissive storage permissions.
+
+A mobile app that stores application shared preferences with world-readable permissions, allowing other apps to read them
+
+**Scenario #4:** Missing jailbreak/root detection.
```

Comment:
I'm surprised to find this under security misconfiguration. Employing a jailbreak/root detection for me is beyond mere configuration. You at least need some additional dependencies, most likely also additional code to make it work properly.